src/main/java/de/tum/in/www1/artemis/repository/CompetencyRepository.java

```diff
@@ -219,34 +202,15 @@ default Competency findByIdElseThrow(long competencyId) {
         return findById(competencyId).orElseThrow(() -> new EntityNotFoundException("Competency", competencyId));
     }
 
-    default Competency findByIdWithLectureUnitsElseThrow(long competencyId) {
-        return findByIdWithLectureUnits(competencyId).orElseThrow(() -> new EntityNotFoundException("Competency", competencyId));
+    default Competency findWithLectureUnitsAndExercisesByIdElseThrow(long competencyId) {
+        return findWithLectureUnitsAndExercisesById(competencyId).orElseThrow(() -> new EntityNotFoundException("Competency", competencyId));
     }
 
-    default Competency findByIdWithExercisesAndLectureUnitsAndProgressForUserElseThrow(long competencyId, long userId) {
-        return findByIdWithExercisesAndLectureUnitsAndProgressForUser(competencyId, userId).orElseThrow(() -> new EntityNotFoundException("Competency", competencyId));
+    default Competency findByIdWithLectureUnitsElseThrow(long competencyId) {
+        return findByIdWithLectureUnits(competencyId).orElseThrow(() -> new EntityNotFoundException("Competency", competencyId));
     }
 
     long countByCourse(Course course);
 
-    /**
-     * Gets all competencies for the given course with the progress of the specified user.
-     * <p>
-     * The query only fetches data related to specified user. Participations for other users are not included.
-     * IMPORTANT: JPA doesn't support JOIN-FETCH-ON statements. To fetch the relevant data we utilize the entity graph annotation.
-     * Moving the ON clauses to the WHERE clause would result in significantly different and faulty output.
-     *
-     * @param courseId the id of the course
-     * @param userId   the id of the user
-     * @return the competencies with the progress of the user
-     */
-    @Query("""
-            SELECT competency
-            FROM Competency competency
-                LEFT JOIN competency.userProgress progress
-                    ON competency.id = progress.competency.id AND progress.user.id = :userId
-            WHERE competency.course.id = :courseId
-            """)
-    @EntityGraph(type = LOAD, attributePaths = { "userProgress" })
-    List<Competency> findByCourseIdWithProgressOfUser(@Param("courseId") long courseId, @Param("userId") long userId);
+    List<Competency> findByCourseId(long courseId);
```

Comment:
why is less data directly queried from the database?

Reply:
(Regarding potential side effects) This method gets only used in the context of these changes.
(Regarding why we don't fetch the data) See my answer to a similar question: https://github.com/ls1intum/Artemis/pull/8052#discussion_r1510290806